### PR TITLE
Add signet network

### PR DESCRIPTION
**Pull request by LLFourn**
Also add a From implementation between nakamoto::Network and bitcoin::Network

### CODE REVIEW

File: common/src/block/checkpoints.rs [hunk 1]
```diff
@@ -26,3 +26,6 @@ pub const TESTNET: &[(u64, &str)] = &[
 
 /// Regtest checkpoints.
 pub const REGTEST: &[(u64, &str)] = &[];
+
+/// Signet checkpoints
```
Comment (vincenzopalazzo) [inline common/src/block/checkpoints.rs:30]:
```suggestion
/// Signet checkpoints.
```

File: common/src/network.rs [hunk 2]
```diff
@@ -43,6 +43,8 @@ pub enum Network {
     Testnet,
     /// Bitcoin regression test net.
     Regtest,
+    /// Bitcoin signet
```
Comment (vincenzopalazzo) [inline common/src/network.rs:46]:
```suggestion
    /// Bitcoin signet.
```

**Review by vincenzopalazzo — APPROVED** [review]
Looks good to me!

Some comments that you can ignore if you want

**Comment by cloudhead** [comment]
Thanks!

**Pull request merged**